File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<String, Integer> map = new HashMap<String, Integer>();
        int output = 0;
        map.put("I", 1);
        map.put("V", 5);
        map.put("X", 10);
        map.put("L", 50);
        map.put("C", 100);
        map.put("D", 500);
        map.put("M", 1000);

        for(int i = 0; i < s.length(); i++){
            String currentChar = Character.toString(s.charAt(i));
            String nextChar = i < (s.length() - 1) ? Character.toString(s.charAt(i+1)) : null;
            if(nextChar != null) {
                if(map.get(currentChar) >= map.get(nextChar)){
                    output += map.get(currentChar);
                } else {
                    output -= map.get(currentChar);
                }
            } else{
                output += map.get(currentChar);
            }  
        }

        return output;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        hashMap = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        num = 0

        for i in range(len(s)):
            if i > 0:
                if hashMap[s[i - 1]] < hashMap[s[i]]:
                    num += hashMap[s[i]] - 2 * hashMap[s[i - 1]]
                else:
                    num += hashMap[s[i]]
            else:
                num += hashMap[s[i]]
                    
        return num